sycl/include/sycl/ext/oneapi/experimental/group_sort.hpp

```diff
@@ -124,16 +126,98 @@ template <typename Group, typename Iter, typename Compare, size_t Extent>
+#else
+  (void)g;
+  (void)key;
+  (void)value;
+  (void)sorter;
+  throw sycl::exception(
+      std::error_code(PI_ERROR_INVALID_DEVICE, sycl::sycl_category()),
+      "Group algorithms are not supported on host device.");
+#endif
+}
+
+template <typename GroupHelper, typename T, typename U, typename Compare,
+          typename Group, std::size_t Extent>
+std::tuple<T, U> sort_key_value_over_group(
+    experimental::group_with_scratchpad<Group, Extent> exec, T key, U value,
+    Compare comp) {
+  return sort_key_value_over_group(
+      exec.get_group(), key, value,
+      default_sorters::group_key_value_sorter<T, U, Compare>(exec.get_memory(),
+                                                             comp));
+}
+
+template <typename T, typename U, typename Group, std::size_t Extent>
+std::tuple<T, U> sort_key_value_over_group(
+    experimental::group_with_scratchpad<Group, Extent> exec, T key, U value) {
+  return sort_key_value_over_group(
+      exec.get_group(), key, value,
+      default_sorters::group_key_value_sorter<T, U>(exec.get_memory()));
+}
+
+template <typename Group, typename T, typename U, std::size_t Extent,
+          std::size_t ElementsPerWorkItem, typename Compare,
+          typename Property = detail::is_blocked>
+void sort_key_value_over_group(
+    experimental::group_with_scratchpad<Group, Extent> exec,
+    sycl::span<T, ElementsPerWorkItem> keys,
+    sycl::span<U, ElementsPerWorkItem> values, Compare comp,
+    Property property = {}) {
+  return experimental::sort_key_value_over_group(
+      exec.get_group(), keys, values,
+      typename experimental::default_sorters::group_key_value_sorter<
+          T, U, Compare, ElementsPerWorkItem>(exec.get_memory(), comp),
+      property);
+}
+
+// TODO: Check for property type
+template <typename Group, typename T, typename U, std::size_t Extent,
+          std::size_t ElementsPerWorkItem,
+          typename Property = detail::is_blocked>
+void sort_key_value_over_group(
+    experimental::group_with_scratchpad<Group, Extent> exec,
+    sycl::span<T, ElementsPerWorkItem> keys,
+    sycl::span<U, ElementsPerWorkItem> values, Property property = {}) {
+  return experimental::sort_key_value_over_group(
+      exec.get_group(), keys, values,
+      typename experimental::default_sorters::group_key_value_sorter<
+          T, U, std::less<>, ElementsPerWorkItem>(exec.get_memory()),
+      property);
+}
+
+// key value sorting
+template <std::size_t ElementsPerWorkItem, typename Group, typename T,
+          typename U, typename ArraySorter,
+          typename Property = detail::is_blocked>
+void sort_key_value_over_group(Group group,
+                               sycl::span<T, ElementsPerWorkItem> keys,
+                               sycl::span<U, ElementsPerWorkItem> values,
+                               ArraySorter array_sorter,
+                               Property property = {}) {
+  array_sorter(group, keys, values, property);
 }
 
```

Comment:
Please, add 3 more API from the Spec document:
```cpp
  template <typename GroupHelper, typename T,
            std::size_t ElementsPerWorkItem,
            typename Properties = ext::oneapi::experimental::empty_properties_t>
  void sort_over_group(GroupHelper gh,
                       sycl::span<T, ElementsPerWorkItem> values,
                       Properties properties = {});

  template <typename GroupHelper, typename T,
            std::size_t ElementsPerWorkItem, typename Compare,
            typename Properties = ext::oneapi::experimental::empty_properties_t>
  void sort_over_group(GroupHelper gh,
                       sycl::span<T, ElementsPerWorkItem> values,
                       Compare comp,
                       Properties properties = {});

  template<typename Group, typename T, std::size_t ElementsPerWorkItem,
           typename Sorter,
           typename Properties = ext::oneapi::experimental::empty_properties_t>
  void sort_over_group(Group g,
                       sycl::span<T, ElementsPerWorkItem> values,
                       Sorter sorter,
                       Properties properties = {});
```

Reply:
Support for the fixed-size arrays is not included in this PR. It's on to do list.